### ISSUE TITLE
FEAT double-layer encryption

### ISSUE THREAD

**Issue by EtherealET** [Shabang-Systems/Condution#77]
When I'm using condition, I'm concerned about the security of my data.
So I need a double-layer encryption feature, which will better protect my data.

**Comment by Jemoka** [comment]
Standard problem -- if we do double-ended encryption, user data gets nuked if they forget their password.

Potential solutions
1. Backup keys owned by the user
2. Some automatic firebase rules that generates and stores the encryption keys and decrypt them only if needed. Emailing the user if they forgot the password to confirm access.
3. Staff-only master root

**Comment by motionvector8** [comment]
If you will add e2e, you will get huge amount of users, including me. Now you are not listed at privacytools.io or f-droid, so it was hard for me to find your app. After e2e release, I think you will be highlighted at many places as trustfull app. Email recovery or/and backup phase will be fair enough (see standardnotes.com)

**Comment by memoryfile** [comment]
There's a lack of FOSS E2E synced to-do list apps right now, so this would be awesome!

**Comment by Jemoka** [comment]
Yup! Thank you for the feedback. We are still working on how the database is going to be deployed to support this; to be honest, we are overly dependent on JSON-based storage (read: Mongo or Firebase) at the moment, and would like to transition away from it; in the same refactor, we will introduce E2E. 

In the meantime self-hosted database is coming together with something called RethinkDB.
